### PR TITLE
[AIRFLOW-4335] Add default num_retries to GCP connection

### PR DESCRIPTION
**Pull request by ryanyuan**
Add default num_retries to GCP connection

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow-4335](https://issues.apache.org/jira/browse/AIRFLOW-4335) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).
  - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by ryanyuan** [comment]
This is to solve #5043

**Comment by ryanyuan** [comment]
@kaxil PTAL

File: airflow/contrib/hooks/bigquery_hook.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def __init__(self,
             gcp_conn_id=bigquery_conn_id, delegate_to=delegate_to)
         self.use_legacy_sql = use_legacy_sql
         self.location = location
+        self.num_retries = self._get_field('num_retries', 5)
```
Comment (potiuk) [inline airflow/contrib/hooks/bigquery_hook.py:63]: I really like that you take values from parameters of Google Connection :)

File: docs/howto/connection/gcp.rst [hunk 2]
```diff
@@ -71,6 +71,12 @@ Scopes (comma separated)
         issue `AIRFLOW-2522
         <https://issues.apache.org/jira/browse/AIRFLOW-2522>`_.
 
+Number of Retries
```
Comment (potiuk) [inline docs/howto/connection/gcp.rst:74]: Love that you remembered about documentation!

**Review by potiuk — APPROVED** [review]
I like the change a lot. I think you need to squash it and rebase and we should be good to go.

File: docs/howto/connection/gcp.rst [hunk 3]
```diff
@@ -71,6 +71,12 @@ Scopes (comma separated)
...
+    Integer, number of times to retry with randomized
+    exponential backoff. If all retries fail, the HttpError
```
Comment (mik-laj) [inline docs/howto/connection/gcp.rst]: Can you use the class role to create a link to the documentation
Example:
```
AAA :class:`~airflow.HttpError` BBBB
```
Reference: 
http://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html

File: airflow/contrib/hooks/bigquery_hook.py (hunk 1 above)
Comment (mik-laj) [reply airflow/contrib/hooks/bigquery_hook.py:63]: If anyone is interested in this subject then the source of the idea is this comment: https://github.com/apache/airflow/pull/5054#discussion_r272843031

**Comment by ryanyuan** [comment]
@potiuk @mik-laj Thanks for your reviews!

Commits squashed and it got rebased.

**Comment by potiuk** [comment]
Unfortunately there was a transient error in Travis (I hope we can get rid of those soon). Looks good, but It would be great if you reword commit message as after squash it has some duplication in.

**Comment by ryanyuan** [comment]
> Unfortunately there was a transient error in Travis (I hope we can get rid of those soon). Looks good, but It would be great if you reword commit message as after squash it has some duplication in.

@potiuk Done.

**Comment by OmerJog** [comment]
@potiuk you'll need to close the ticket and target it to 2.0.0  :)
Do we want this for 1.10.4 ? If so it will require another PR against the V1.10 branch as it require UI changes.

**Comment by potiuk** [comment]
Yeah. If I only could :). Waiting for the rights to be able to do so ;)

**Comment by potiuk** [comment]
Also I think decision on what to merge for 1.10.4 will be done by release manager :)

**Pull request merged**